trie/secure_trie.go

```diff
@@ -214,6 +214,24 @@ func (t *StateTrie) GetKey(shaKey []byte) []byte {
 	return t.db.Preimage(common.BytesToHash(shaKey))
 }
 
+func (t *StateTrie) AccessList() map[string][]byte {
+	return t.trie.AccessList()
+}
+
+func (t *StateTrie) CommitAndObtainAccessList(collectLeaf bool) (common.Hash, *trienode.NodeSet, map[string][]byte, error) {
```

Comment:
We should not have such methods. Instead, simply just not nuke out the access list after a commit. It's wonky to have these methods injected all over.

Reply:
Resolved this in [3e305e6f2a44a4de5eb4758ff778b58fa6e405c9](https://github.com/ethereum/go-ethereum/pull/29719/commits/3e305e6f2a44a4de5eb4758ff778b58fa6e405c9) .  Pulled out what seems like a simple one-liner change into its own [PR](https://github.com/ethereum/go-ethereum/pull/29719).